Mathlib/RingTheory/Ideal/IsPrimary.lean

```diff
@@ -14,28 +15,44 @@ A proper ideal `I` is primary iff `xy ∈ I` implies `x ∈ I` or `y ∈ radical
 
 - `Ideal.IsPrimary`
 
+## Implementation details
+
+Uses a specialized phrasing of `Submodule.IsPrimary` to have better API-piercing usage.
+
 -/
 
 namespace Ideal
 
 variable {R : Type*} [CommSemiring R]
 
 /-- A proper ideal `I` is primary iff `xy ∈ I` implies `x ∈ I` or `y ∈ radical I`. -/
-def IsPrimary (I : Ideal R) : Prop :=
+protected def IsPrimary (I : Ideal R) : Prop :=
   I ≠ ⊤ ∧ ∀ {x y : R}, x * y ∈ I → x ∈ I ∨ y ∈ radical I
 
-theorem IsPrime.isPrimary {I : Ideal R} (hi : IsPrime I) : IsPrimary I :=
+lemma isPrimary_iff {I : Ideal R} :
+    I.IsPrimary ↔ I ≠ ⊤ ∧ ∀ {x y : R}, x * y ∈ I → x ∈ I ∨ y ∈ radical I :=
+  Iff.rfl
+
+/-- An ideal `I : Ideal R` is primary iff it is primary as a submodule. This links
+`Ideal.IsPrimary` with `Submodule.IsPrimary`. -/
+lemma isPrimary_iff' {I : Ideal R} :
+    I.IsPrimary ↔ Submodule.IsPrimary I := by
+  rw [isPrimary_iff, Submodule.isPrimary_iff, forall_comm]
+  simp only [mul_comm, mem_radical_iff, and_congr_right_iff,
+    ← Submodule.ideal_span_singleton_smul, smul_eq_mul, mul_top, span_singleton_le_iff_mem]
```

Comment:
I would prefer if `isPrimary_iff'` was definitional. Because ideals are definitionally submodules.

- I think it will be less confusing
- Any API for `Submodule.IsPrimary` can be used directly for `Ideal.IsPrimary`. (The latter could be an `abbrev`.)